plugin-annotation/src/main/java/com/mapbox/mapboxsdk/plugins/annotation/AnnotationManager.java

```diff
@@ -370,12 +372,13 @@ public void onDestroy() {
     private void initializeSourcesAndLayers(GeoJsonOptions geoJsonOptions) {
         geoJsonSource = coreElementProvider.getSource(geoJsonOptions);
         layer = coreElementProvider.getLayer();
-
         style.addSource(geoJsonSource);
-        if (belowLayerId == null) {
-            style.addLayer(layer);
-        } else {
+        if (belowLayerId != null) {
             style.addLayerBelow(layer, belowLayerId);
+        } else if (aboveLayerId != null) {
+            style.addLayerAbove(layer, aboveLayerId);
+        } else {
+            style.addLayer(layer);
```

Comment:
The code appears to imply that `aboveLayerId` is ignored when `belowLayerId` is set. Do you think it would make sense to have a behavior noticeable to the developer to inform them that one parameter is being ignored?

In a Kotlinified version of this file (re. https://github.com/maplibre/maplibre-native/pull/1255) I would expect both of them to be optional parameters and an exception to be thrown if both are provided.